test/extensions/filters/http/cache/cache_headers_utils_test.cc

```diff
@@ -0,0 +1,320 @@
+#include <chrono>
+#include <string>
+#include <vector>
+
+#include "envoy/common/time.h"
+
+#include "common/http/header_map_impl.h"
+
+#include "extensions/filters/http/cache/cache_headers_utils.h"
+
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+namespace {
+
+struct TestRequestCacheControl : public RequestCacheControl {
+  TestRequestCacheControl(bool must_validate, bool no_store, bool no_transform, bool only_if_cached,
+                          OptionalDuration max_age, OptionalDuration min_fresh,
+                          OptionalDuration max_stale) {
+    must_validate_ = must_validate;
+    no_store_ = no_store;
+    no_transform_ = no_transform;
+    only_if_cached_ = only_if_cached;
+    max_age_ = max_age;
+    min_fresh_ = min_fresh;
+    max_stale_ = max_stale;
+  }
+};
+
+struct TestResponseCacheControl : public ResponseCacheControl {
+  TestResponseCacheControl(bool must_validate, bool no_store, bool no_transform, bool no_stale,
+                           bool is_public, OptionalDuration max_age) {
+    must_validate_ = must_validate;
+    no_store_ = no_store;
+    no_transform_ = no_transform;
+    no_stale_ = no_stale;
+    is_public_ = is_public;
+    max_age_ = max_age;
+  }
+};
+
+struct RequestCacheControlTestCase {
+  absl::string_view cache_control_header;
+  TestRequestCacheControl request_cache_control;
+};
+
+struct ResponseCacheControlTestCase {
+  absl::string_view cache_control_header;
+  TestResponseCacheControl response_cache_control;
+};
+
+class RequestCacheControlTest : public testing::TestWithParam<RequestCacheControlTestCase> {
+public:
+  static auto getTestCases() {
+    // clang-format off
+    static const RequestCacheControlTestCase test_cases[] = {
```

Comment:
does the TestResponseCacheControl actually get called? If so, this is a fiasco violation. If it doesn't, might as well just remove it.

See https://github.com/envoyproxy/envoy/blob/master/STYLE.md and search for 'fiasco' for context.

If this strictly used POD-initialization then it gets initialized at the compile time and there is no problem.

Reply:
I will replace it with CONSTRUCT_ON_FIRST_USE and use a vector.